pkg/apiserver/rest/dualwriter.go

```diff
@@ -3,12 +3,12 @@ package rest
 import (
 	"context"
 
+	"github.com/grafana/grafana/pkg/infra/log"
```

Comment:
i think this is going to fail lint because it's importing from `grafana/grafana`

Reply:
@marefr since you have been working on API server o11y as well, do you have any thoughts about how to handle this?

Reply:
hm, I'm wondering if this is something we should support, though, since this has been the default logger so far? I'm happy to use `klog` as well, I'm just not sure if we have reached some sort of consensus there already.

Reply:
@leonorfmartins it's definitely something we could support. i think it would require either moving `infra/log` to a sub-module (new or existing), or moving the dual writer out of a sub-module and back into `grafana/grafana`

Reply:
Hmm from where are dual writer used/initialized, within the apiserver package? If not, we could maybe extract the logger interface to begin with? And then up to the apiserver cli/command to provide the implementation? It's an interesting discussion what logger interface/package to standardize on, dskit/other grafana projects uses github.com/go-kit/log for example.

We also have this [interface in plugin sdk](https://github.com/grafana/grafana-plugin-sdk-go/blob/63f03d325903aad46c666a0412e32d2bfceedbb0/backend/log/log.go#L22), but probably not a good idea to use outside plugins.

klog doesn't support context I guess?

> my guess is that we are probably going to need to move infra/log into a sub-module anyway, even if it's not for this specific use case. thoughts?

Maybe, but coming back to what I wrote earlier, we at least need a logger interface to standardize on, but perhaps not moving the whole package.

Reply:
> klog doesn't support context I guess?

https://pkg.go.dev/k8s.io/klog/v2#pkg-overview it looks like it supports contextual and structured logging

> a logger interface to standardize on

good point. using the existing grafana interface makes sense to me, but i'm also wondering if `log/slog` would be worth considering since it's now included with go?

Reply:
> good point. using the existing grafana interface makes sense to me, but i'm also wondering if log/slog would be worth considering since it's now included with go?

Yeah, could be. Maybe something to discuss in backend engineering council?

Reply:
For now I can use klog as it seems to be the log used in the apiserver package, since we have some urgency to have _some log_ which we can use with contexts. However, it would be worth to just pick one. I can bring that discussion to the next council.

Reply:
sounds great to me, thanks!